### PR TITLE
Walkthrough setup

### PR DESCRIPTION
**Pull request by deathbearbrown**
Walkthroughs are here and boy are they... ... well they aren't the most attractive thing but they have :sparkles: potential :sparkles: 

This PR includes the following:
- A new fancy footer and modal!  You can probably close PR #339 if you merge this in.
- A walkthrough menu for selecting walkthroughs
- some redux reducers for storing the walkthrough state
- a new hints box that needs some css love but hey, it's ok for now
- the ugliest error messaging you've ever seen
- a lot of Bill Murray.  There are 2 example walkthroughs, that exist in /walkthrough/files.   Am I in love with this folder structure? nope.
- I am not loading the walkthrough json through "thunks", I am just loading them through require since they exist inside the app. I thought about this and I don't see any reason you might host them outside of the app.

walkthrough

**This PR includes:**
- [x] Tests
- [x] functional comments
- [x] high level description of any new components

**Steps to functionally test this:**
- click on the WALKTHROUGHS link in the footer
- select a walkthrough
- follow the prompted steps, or don't follow them and get errors. It's your life and I'm not gonna tell you how to live it.  

The validator is very forgiving. It's only going to check that the things you asked for in the walkthrough spec exist, but if you do extra fancy things like change colors, or fonts or add 17 line marks, it's not going to yell at you as long as you meet the requirement of the prompts. That's by design.

### CODE REVIEW

**Comment by deathbearbrown** [comment]
I think this needs some css love and a good functional walkthrough with feedback, but just in case you're feeling merge-happy, please wait until I do some css cleanup and squash these commits.  :)

**Comment by deathbearbrown** [comment]
ok I cleaned up the css and squashed this so it's good to review/merge if it looks ok & travis doesn't die again (I could have sworn we already had lodash in package.js)

File: package.json [hunk 1]
```diff
@@ -62,6 +62,7 @@
     "karma-sourcemap-loader": "^0.3.6",
     "karma-spec-reporter": "0.0.23",
     "karma-webpack": "^1.7.0",
+    "lodash": "^4.11.1",
```
Comment (arvind) [inline package.json]: Does webpack do tree-shaking? If not, perhaps we want to just include the subpackages for the particular functions we care about to reduce vendor dependency size?

File: src/js/components/Footer.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+'use strict';
+var React = require('react'),
+    connect = require('react-redux').connect,
+    Modal = require('react-modal'),
+    Walkthroughs = require('./walkthrough/Menu');
+
+
+var Footer = connect()(React.createClass({
```
Comment (arvind) [inline src/js/components/Footer.jsx]: Why the empty `connect`? Is that necessary for the modal, or leftover code?

File: src/js/components/walkthrough/hints.jsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
+'use strict';
```
Comment (arvind) [inline src/js/components/walkthrough/hints.jsx]: I think this should be `Hints.jsx` (capital `H`)?

File: src/js/util/walkthrough-utils.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+/**
+ * Get an array of all elements with a certain type
+ * Wraps lodash filter
+ * @param  {string} elemType - a string that will match the key in each state element
+ * @param  {Array} stateArray - an array of objects to test against
+ * @param  {string} key - key to filter by
+ * @returns {[type]}            [description]
+ */
+function filterBy(elemType, stateArray, key) {
+  return _.filter(stateArray, function(o) {
```
Comment (arvind) [inline src/js/util/walkthrough-utils.js]: Does a regular `stateArray.filter` suffice here? Reduces our dependency on lodash, if we switch to their modular dependencies.

File: src/js/util/walkthrough-utils.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+  return propertyObj;
+}
+
+/**
+ * Get an array of all elements with a certain type
+ * Wraps lodash filter
+ * @param  {string} elemType - a string that will match the key in each state element
+ * @param  {Array} stateArray - an array of objects to test against
+ * @param  {string} key - key to filter by
+ * @returns {[type]}            [description]
```
Comment (arvind) [inline src/js/util/walkthrough-utils.js]: Missing returns jsdoc.

File: src/js/walkthrough/files/example-different.json [hunk 6]
```diff
@@ -0,0 +1,77 @@
+{
+  "title": "Cool times",
+  "image": "https://placeholdit.imgix.net/~text?txtsize=33&txt=150%C3%97150&w=150&h=150",
+  "alt_text": "an image",
+  "steps": [{
+    "id": 1,
+    "title": "Step 1: Add a SYMBOL",
+    "text": "Click on the SYMBOL button in the toolbar to add a new Rect to the canvas",
+    "image": "http://fillmurray.com/g/150/150",
```
Comment (arvind) [inline src/js/walkthrough/files/example-different.json:9]: Oh my god, this is amazing.

**Comment by arvind** [comment]
Functional +1. This is fantastic. Some minor comments throughout the code.

File: src/js/components/Footer.jsx (hunk 2 above)
Comment (deathbearbrown) [inline src/js/components/Footer.jsx]: left overs! I will clean up

File: src/js/util/walkthrough-utils.js (hunk 4 above)
Comment (deathbearbrown) [inline src/js/util/walkthrough-utils.js]: I can switch it up

**Comment by deathbearbrown** [comment]
Removed all the lodash! :O  Made all the updates

File: src/js/util/walkthrough-utils.js [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+ * @param  {Object} match - the matching element
+ * @param  {Object} elem - an object to match against
+ * @param  {String} type - the type in the store eg - Marks, Scales, Guide
+ * @returns {Boolean} Success or failure of the test
+ */
+function matchPropertyExistence(match, elem, type) {
+  // get the container of the properties by type
+  var matchProps = getProperties(match, type),
+      elemProps = getProperties(elem, type);
+  return _.isMatch(elemProps, matchProps);
```
Comment (arvind) [inline src/js/util/walkthrough-utils.js]: Still some lodash here! Is it simple enough to write ourselves, or we could [require lodash-ismatch](https://www.npmjs.com/package/lodash.ismatch)?

Comment (deathbearbrown) [inline src/js/util/walkthrough-utils.js]: :C yeah I just caught that

**Pull request merged**